src/View/Sidebar/AbstractMountableRow.vala

```diff
@@ -0,0 +1,433 @@
+/* DeviceRow.vala
+ *
+ * Copyright 2021 elementary LLC. <https://elementary.io>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ *
+ * Authors : Jeremy Wootten <jeremy@elementaryos.org>
+ */
+
+public abstract class Sidebar.AbstractMountableRow : Sidebar.BookmarkRow, SidebarItemInterface {
+    protected static Gtk.CssProvider devicerow_provider;
+
+    protected Gtk.Stack mount_eject_stack;
+    protected Gtk.Revealer mount_eject_revealer;
+    protected Gtk.Spinner mount_eject_spinner;
+
+    protected bool valid = true;
+    public string? uuid { get; set construct; }
+    public Drive? drive { get; set construct; }
+    public Volume? volume { get; set construct; }
+    public Mount? mount { get; set construct; }
+
+    protected bool _mounted = false;
+    public bool mounted {
+        get {
+            return _mounted;
+        }
+
+        set {
+            _mounted = value;
+            can_eject = _mounted && (mount.can_unmount () || mount.can_eject ());
+            mount_eject_revealer.reveal_child = _mounted && _can_eject;
+        }
+    }
+
+    protected bool _can_eject = false;
+    public bool can_eject {
+        get {
+            return _can_eject;
+        }
+
+        set {
+            _can_eject = !permanent && value;
+             mount_eject_revealer.reveal_child = _can_eject && _mounted;
+        }
+    }
+
+    private bool _working = false;
+    public bool working {
+        get {
+            return _working;
+        }
+
+        set {
+            if (!valid) {
+                return;
+            }
+
+            _working = value;
+
+            if (value) {
+                mount_eject_revealer.reveal_child = true;
```

Comment:
It looks like you set this again on line 83 so I guess this really won't end up doing anything right?